src/main/javascript/crypto/src/Authenticator.ts

```diff
@@ -52,106 +45,31 @@ const ALPHA_CONFIG = {
-        // let tokenType = Negotiator.getTokenType(tokenObj.tokenClass);
-        // 2. Trace from its TokenScript which website has the needed data object
-        // if (tokenType.attestationOrigin) { // always return true in Devcon project,
-        // unless DevCon changed their tokenscript and moved all tickets to the contract
-
-        this.signedTokenBlob = tokenObj.ticketBlob;
-        this.magicLink = tokenObj.magicLink;
-        this.email = tokenObj.email;
-        this.signedTokenSecret = tokenObj.ticketSecret;
-        this.attestationOrigin = tokenObj.attestationOrigin;
-        this.authResultCallback = authResultCallback;
-        this.getIdentifierAttestation();
-
-    }
-
-    /*
-     *  - Since this token depends on identifier attestation, continue to open iframe to attestation.id who needs to provide the proof
-     */
-    getIdentifierAttestation() {
-        logger(DEBUGLEVEL.HIGH,'getIdentifierAttestation. create iframe with ' + this.attestationOrigin);
-
-        // attach postMessage listener and wait for attestation data
-        this.attachPostMessageListener(this.postMessageAttestationListener.bind(this));
-        const iframe = document.createElement('iframe');
-        this.iframe = iframe;
-        iframe.src = this.attestationOrigin;
-        iframe.style.width = '800px';
-        iframe.style.height = '700px';
-        iframe.style.maxWidth = '100%';
-        iframe.style.background = '#fff';
-        let iframeWrap = document.createElement('div');
-        this.iframeWrap = iframeWrap;
-        iframeWrap.setAttribute('style', 'width:100%;min-height: 100vh; position: fixed; align-items: center; justify-content: center;display: none;top: 0; left: 0; background: #fffa');
-        iframeWrap.appendChild(iframe);
-
-        document.body.appendChild(iframeWrap);
-    }
-
-    async getUseTicket(
+    // TODO: Pass in Ticket schema object
+    static async getUseTicket(
         // userKey: KeyPair,
         ticketSecret: bigint,
         attestationSecret: bigint,
         base64ticket: string,
         base64attestation: string,
         base64attestationPublicKey: string,
-        base64senderPublicKey: string
+        base64senderPublicKey: string,
+        useOldTicketSchema: boolean = false
     )
     {
-        let ticket: Ticket;
+        let ticket: Ticket|TicketTNCompat;
         let att: SignedIdentifierAttestation;
 
         // let ticket: Ticket = Ticket.fromBase64(base64ticket, KeyPair.fromPublicHex(base64senderPublicKey));
         try {
-            ticket = Ticket.fromBase64(base64ticket,{"6": KeyPair.publicFromBase64(base64senderPublicKey)});
+            if (useOldTicketSchema){
+                ticket = TicketTNCompat.fromBase64(base64ticket,KeyPair.publicFromBase64(base64senderPublicKey));
+            } else {
+                ticket = Ticket.fromBase64(base64ticket,{"6": KeyPair.publicFromBase64(base64senderPublicKey)});
+            }
+
```

Comment:
Better remove hardcoded conferenceID "6" and receive keysArray as argument, like here
https://github.com/TokenScript/attestation/blob/ticket_validator/src/main/javascript/crypto/src/Authenticator.ts#L682-L710

Reply:
this code can be updated after merge with main, because its related to the Ticket